src/Mutator/Util/Mutator.php

```diff
@@ -48,6 +49,11 @@ public function __construct(MutatorConfig $config)
         $this->config = $config;
     }
 
+    public static function getDefinition(): ?Definition
```

Comment:
Note that this class (or extracted interface) will be an extension point. Thus, users will have to implement this method, which I don't really think they should.

Definitions (in any way) are definitely useful for Infection itself, but not for the custom mutators (Imagine you are creating a custom mutator Infection should use in your project).

Probably, this method should be on the higher level, in another base class that Infection core's mutator would extend.

I just see when the developers are not clear understand what is the `?Definition` is and will just return `null`. And even if they understand, there is no point to return any Definition from a custom mutator (from the first look).

Reply:
If we provide the base abstract class as an extension point, then the problem is taken care of. If it's an interface, we can not include this from the interface but include it in our mutators still.

So I don't think it will a problem, and since it's not entirely clear yet how we are opening up this bit I wouldn't worry too much about it.

Reply:
This 100% will be a problem, but let's think about it later.